src/containers/App.js

```diff
@@ -0,0 +1,101 @@
+
+
+// src/containers/App.js
+
+import React, { Component } from 'react';
+import { ALL_TODOS, ACTIVE_TODOS, COMPLETED_TODOS } from '../constants';
+
+import Header from '../components/Header';
+import Todos from '../components/Todos';
+import Footer from '../components/Footer';
+
+class App extends Component {
+	constructor(props) {
+		super(props);
+		
+		this.state = {
+			show: ALL_TODOS,
+			items: [
+				{ label: 'item 1', done: 0 },
+				{ label: 'item 2', done: 0 },
+				{ label: 'item 3', done: 0 },
+				{ label: 'item 4', done: 0 },
+			],
+		};
+	}
+	
+	setFilter(show) {
+		this.setState({ show: show });
+	}
+	
+	addEntry(label) {
+		// I was about to add code to stop it adding duplicate labels (since we use the labels as keys)
+		// but I tested and found it doesn't add them
+		// this has to be because we're using the label as the key
+		// but I wonder if there are unwanted side-effects of relying on this, rather than explicitly coding to avoid the duplicates?
+		
+		if (!this.state.items.find(item => item.label === label)) {
+			this.setState(old => ({
+				items: [
+					...old.items,
+					{
+						label: label,
+						done: 0,
+					},
+				],
+			}));
+		}
+	}
+	
+	setItemCheck(label, done) {
+		this.setState(old => {
+			const newItems = [...old.items];

```

Comment:
Great use of spread operators to ensure no data is mutated.  That is what React + Redux is all immutability.